distributed/protocol/cupy.py

```diff
@@ -0,0 +1,39 @@
+"""
+Efficient serialization GPU arrays.
+"""
+import cupy
+from .serialize import dask_serialize, dask_deserialize
+
+
+@dask_serialize.register(cupy.ndarray)
+def serialize_cupy_ndarray(x):
+    # TODO: handle non-contiguous
+    # TODO: Handle order='K' ravel
+    # TODO: 0d
+
+    if x.flags.c_contiguous or x.flags.f_contiguous:
+        strides = x.strides
+        data = x.ravel()  # order='K'
+    else:
+        x = cupy.ascontiguousarray(x)
+        strides = x.strides
+        data = x.ravel()
+
+    dtype = (0, x.dtype.str)
+
+    header = x.__cuda_array_interface__.copy()
+    header['lengths'] = (x.nbytes,)  # one per stride
```

Comment:
Ideally we wouldn't need this.  Dask should populate this automatically if the `nbytes` function works on the bytes-like objects returned as frames.